### PR TITLE
5956 reporting on datasources

### PR DESCRIPTION
**Pull request by dannysmyda**

### CODE REVIEW

**Comment by eugene7646** [comment]
@dannysmyda Codacy is complaining about one "if" statement.

File: Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+package org.sleuthkit.autopsy.report;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Configuration parameters for general report modules
+ */
+public class GeneralReportSettings implements Serializable {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java:28]: Is making this Serializable looking towards the future?

File: Core/src/org/sleuthkit/autopsy/report/GeneralReportModule.java [hunk 2]
```diff
@@ -25,10 +25,36 @@
...
+     * Report should go into baseReportDir + getRelativeFilePath().
      * @param progressPanel panel to update the report's progress with
      */
     public void generateReport(String baseReportDir, ReportProgressPanel progressPanel);
 
+    /**
+     * Determines if the module supports report generation on a subset of data
+     * sources in a case. Defaults to false. Modules that override this to true
+     * should implement all generateReport overloads. The data source selections
+     * are stored in the GeneralReportSettings instance.
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/report/GeneralReportModule.java:37]: "Modules that override this to true should implement all generateReport overloads."

See comments in Jira.

File: Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    private static final long serialVersionUID = 1L;
+    
+    private List<Long> dataSourcesToProcess;
+    private String reportDirectoryPath;
+    
+    public List<Long> getDataSourcesToProcess() {
```
Comment (rcordovano) [inline Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java:34]: How about getSelectedDataSources/setSelectedDataSources?

File: Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java (hunk 1 above)
Comment (dannysmyda) [reply Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java:28]: No, from what I remember all settings are first assembled -> then serialized -> then deserialized -> then given to the modules. Going to disk is a necessary step in the pipeline.

File: Core/src/org/sleuthkit/autopsy/geolocation/datamodel/GeoPath.java [hunk 4]
```diff
@@ -126,7 +126,7 @@ final void addToPath(Waypoint point) {
      *
      * @return
      */
-    final BlackboardArtifact getArtifact() {
+    public final BlackboardArtifact getArtifact() {
```
[the remark this answers is not in the document]
Comment (dannysmyda) [reply Core/src/org/sleuthkit/autopsy/geolocation/datamodel/GeoPath.java:129]: The KML report pulls routes, tracks and waypoints from certain artifacts across the entire case. By making this public, I was able to get at the underlying artifact to filter the data source. However, I see that Track.getTracks(), for example, accepts a list of content objects to filter by. Routes and Waypoints don't have this, but maybe a better idea is to add methods in these classes.

File: Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java (hunk 3 above)
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java:34]: Not really - selection can also happen via command line args or entries read from a configuration file. I was thinking that process sounded a little heavy-duty for reporting. It's not a big deal though.

File: Core/src/org/sleuthkit/autopsy/geolocation/datamodel/GeoPath.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/geolocation/datamodel/GeoPath.java:129]: Moving this conversation to Jira.

File: Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java (hunk 1 above)
Comment (rcordovano) [reply Core/src/org/sleuthkit/autopsy/report/GeneralReportSettings.java:28]: OK.

File: Core/src/org/sleuthkit/autopsy/corecomponents/CheckBoxJList.java [hunk 5]
```diff
@@ -16,7 +16,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.sleuthkit.autopsy.geolocation;
+package org.sleuthkit.autopsy.corecomponents;
```
[the remark this answers is not in the document]
Comment (dannysmyda) [reply Core/src/org/sleuthkit/autopsy/corecomponents/CheckBoxJList.java:19]: Okay, I'll change this. Thank you

**Pull request merged**